airflow/providers/google/cloud/operators/translate_speech.py

```diff
@@ -179,12 +179,13 @@ def execute(self, context: Context) -> dict:
                 model=self.model,
             )
             self.log.info("Translated output: %s", translation)
-            FileDetailsLink.persist(
-                context=context,
-                task_instance=self,
-                uri=self.audio["uri"][5:],
-                project_id=self.project_id or translate_hook.project_id,
-            )
+            if self.audio.uri:
+                FileDetailsLink.persist(
+                    context=context,
+                    task_instance=self,
+                    uri=self.audio.uri[5:],
+                    project_id=self.project_id or translate_hook.project_id,
+                )
```

Comment:
Could you please move this block into the beginning of the method as it is done for the  `CloudSpeechToTextRecognizeSpeechOperator`?
The reason is that the link appears in the Airflow UI immediately after calling the `FileDetailsLink.persist(...)` method. That's why even if something bad happens, the link will be populated already.